libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -333,6 +343,30 @@ public void onQueryCurrentProgressForUploadingMedia() {
                         updateMediaProgress();
                     }
                 },
+                new OnSetFeaturedImageListener() {
+                    @Override
+                    public void onSetFeaturedImageButtonClicked(int mediaId) {
+                        if (mediaId == mFeaturedImageId) {
+                            // nothing special to do, trying to set the image that's already set as featured
+                            return;
+                        }
+
+                        if (mediaId == 0) {
+                            // user tries to clear the featured image setting
+                            setFeaturedImage(mediaId);
+                            return;
+                        }
+
+                        if (mFeaturedImageId == 0) {
+                            // current featured image is not set so, go ahead and set it to the provided one
+                            setFeaturedImage(mediaId);
```

Comment:
Within `EditorFragmentAbstract` `mFeaturedImageSupported` is being persisted via the `savedInstanceState` bundle. We may need to persist `mFeaturedImageId`. Have you tried rotating the device while a `mFeaturedImageId` is set, attempt to set the same media id, and see if the function returns? Let me know what that is like. I will do another test of this PR tomorrow.

Reply:
@jd-alexander, `mFeaturedImageId` was added to `savedInstanceState` within `GutenbergEditorFragment` in https://github.com/wordpress-mobile/WordPress-Android/pull/14503/commits/f57eeaf8703a2ef816cc98949a54b99ab6432584, which I believe fixes any issues with it being reset with screen rotation. Would it be better practice to do this in `EditorFragmentAbstract` rather than `GutenbergEditorFragment`?

Reply:
Ah, I didn't notice that. Thanks for pointing it out.